web/middlewares.go

```diff
@@ -0,0 +1,29 @@
+package web
+
+import (
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+	"github.com/sirupsen/logrus"
+	"github.com/trento-project/trento/web/services"
+)
+
+func EulaMiddleware(settings services.SettingsService) gin.HandlerFunc {
+	return func(c *gin.Context) {
+		eulaAccepted, err := settings.IsEulaAccepted()
+		if err != nil {
+			logrus.Error(err)
+			c.HTML(http.StatusInternalServerError, "error.html.tmpl", gin.H{"Error": "EULA Error :-("})
+			c.Abort()
```

Comment:
Do we need these `aborts` here? I would think with the code in hand that the `c.HTML` is already returning

Reply:
Forget what I said, it looks a normal practice to stop the next handlings to be executed